### PR TITLE
Support for SFAuthenticationSession in iOS 11.

### PR DESCRIPTION
**Pull request by WilliamDenniss**
Added support for SFAuthenticationSession to enable SSO with AppAuth in iOS 11.

https://twitter.com/rmondello/status/884458464470343682
https://developer.apple.com/documentation/safariservices/sfauthenticationsession?changes=latest_beta&language=objc

### CODE REVIEW

**Comment by WilliamDenniss** [comment]
NB. CI is failing as it doesn't yet support the iOS 11 beta 3.

File: Source/iOS/OIDAuthorizationUICoordinatorIOS.m [hunk 1]
```diff
@@ -45,6 +46,7 @@ @implementation OIDAuthorizationUICoordinatorIOS {
   BOOL _authorizationFlowInProgress;
   __weak id<OIDAuthorizationFlowSession> _session;
   __weak SFSafariViewController *_safariVC;
+   SFAuthenticationSession *_authenticationVC;
```
Comment (XiangtianDai) [inline Source/iOS/OIDAuthorizationUICoordinatorIOS.m]: This line would require using iOS 11 SDK (i.e., Xcode 9).

Comment (WilliamDenniss) [inline Source/iOS/OIDAuthorizationUICoordinatorIOS.m]: Thanks! I added some #ifdefs to support both iOS 11 and 10 as the base SDK. PTAL.

File: Source/iOS/OIDAuthorizationUICoordinatorIOS.m [hunk 2]
```diff
@@ -71,23 +73,40 @@ - (nullable instancetype)initWithPresentingViewController:
...
+  NSURL *requestURL = [request authorizationRequestURL];
+  
+  if (@available(iOS 11.0, *)) {
+    NSString *redirectScheme = request.redirectURL.scheme;
+    SFAuthenticationSession* authenticationVC =
+        [[SFAuthenticationSession alloc] initWithURL:requestURL
+                                   callbackURLScheme:redirectScheme
+                                   completionHandler:^(NSURL * _Nullable callbackURL,
+                                                       NSError * _Nullable error) {
+      [_session resumeAuthorizationFlowWithURL:callbackURL];
```
Comment (XiangtianDai) [inline Source/iOS/OIDAuthorizationUICoordinatorIOS.m]: You probably want to set `_authenticationVC` to `nil` either here or in `cleanUp`.

**Comment by WilliamDenniss** [comment]
Merged into a new beta branch: `ios11beta`.  This branch will be merged to master once the SDK goes GM.

**Comment by chanthydara** [comment]
#360

**Comment by crowe2134** [comment]
cbc73f740ebd

**Comment by crowe2134** [comment]
cbc73

**Comment by crowe2134** [comment]
c

**Pull request closed without merging**